### PR TITLE
chore(react-integration): resolved linting errors

### PR DESCRIPTION
**Pull request by evwilkin**
**What**: Towards #3592 - resolves existing linting issues for react-integration

### CODE REVIEW

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ApplicationLauncherDemo/ApplicationLauncherFavoritesDemo.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export class ApplicationLauncherFavoritesDemo extends React.Component {
       isOpen
     });
   };
-  onSelect = event => {
+  onSelect = () => {
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ApplicationLauncherDemo/ApplicationLauncherFavoritesDemo.tsx]: As a demo, it would be good to show what args can be are expected here

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/BackgroundImageDemo/BackgroundImageDemo.tsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
 import { BackgroundImage, BackgroundImageProps, BackgroundImageSrc } from '@patternfly/react-core';
+/* eslint-disable @typescript-eslint/no-var-requires, camelcase */
 const pfbg_1200 = require('../LoginPageDemo/images/pfbg_1200.jpg');
 const pfbg_768 = require('../LoginPageDemo/images/pfbg_768.jpg');
 const pfbg_768_2x = require('../LoginPageDemo/images/pfbg_768@2x.jpg');
 const pfbg_576 = require('../LoginPageDemo/images/pfbg_576.jpg');
 const pfbg_576_2x = require('../LoginPageDemo/images/pfbg_576@2x.jpg');
+/* eslint-enable @typescript-eslint/no-var-requires */
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/BackgroundImageDemo/BackgroundImageDemo.tsx]: I'm wondering why we have the no-var-requires rule? We appear to comment it out everywhere it's being used.

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/BrandDemo/BrandDemo.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { Brand } from '@patternfly/react-core';
-import pfLogo from './images/pfLogo.svg';
+const pfLogo = './images/pfLogo.svg';
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/BrandDemo/BrandDemo.tsx]: Was this change intentional? Perhaps fixing a bug unrelated to the linter?

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ContextSelectorDemo/ContextSelectorDemo.tsx [hunk 4]
```diff
@@ -53,7 +53,7 @@ export class ContextSelectorDemo extends React.Component<{}, ContextSelectorStat
       this.setState({ searchValue: value });
     };
 
-    this.onSearchButtonClick = event => {
+    this.onSearchButtonClick = () => {
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ContextSelectorDemo/ContextSelectorDemo.tsx]: As a demo, it would be good to show what args can be are expected here

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/DropdownDemo/DropdownDemo.tsx [hunk 5]
```diff
@@ -81,16 +82,18 @@ export class DropdownDemo extends React.Component<{}, DropdownState> {
         isCogOpen
       });
     };
-    this.onCogSelect = event => {
+    this.onCogSelect = () => {
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/DropdownDemo/DropdownDemo.tsx]: As a demo, it would be good to show what args can be are expected here

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/FormSelectDemo/FormSelectDemo.tsx [hunk 6]
```diff
@@ -52,11 +52,11 @@ export class FormSelectDemo extends Component<{}, FormSelectState> {
     ];
   }
 
-  onChange = (value: string, event: any) => {
+  onChange = (value: string) => {
     this.setState({ value });
   };
 
-  onChangeValidatedSelect = (value: string, event: any) => {
+  onChangeValidatedSelect = (value: string) => {
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/FormSelectDemo/FormSelectDemo.tsx]: As a demo, it would be good to show what args can be are expected here

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/LoginPageDemo/LoginPageDemo.tsx [hunk 7]
```diff
@@ -1,10 +1,12 @@
 import * as React from 'react';
+/* eslint-disable @typescript-eslint/no-var-requires, camelcase */
 const brandImg = require('./images/brandImgColor.svg') as string;
 const pfbg_1200 = require('./images/pfbg_1200.jpg');
 const pfbg_768 = require('./images/pfbg_768.jpg');
 const pfbg_768_2x = require('./images/pfbg_768@2x.jpg');
 const pfbg_576 = require('./images/pfbg_576.jpg');
 const pfbg_576_2x = require('./images/pfbg_576@2x.jpg');
+/* eslint-enable @typescript-eslint/no-var-requires */
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/LoginPageDemo/LoginPageDemo.tsx]: Again, questioning the no-var-requires rule

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/OverflowMenuDemo/OverflowMenuDemo.tsx [hunk 8]
```diff
@@ -167,7 +167,7 @@ export class OverflowMenuDemo extends React.Component {
     });
   };
 
-  onPersistSelect = event => {
+  onPersistSelect = () => {
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/OverflowMenuDemo/OverflowMenuDemo.tsx]: As a demo, it would be good to show what args can be are expected here

File: packages/patternfly-4/react-integration/scripts/junit2html.js [hunk 9]
```diff
@@ -1,6 +1,8 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
 const fs = require('fs');
 const path = require('path');
 const { getHTMLReport } = require('./htmlReport');
+/* eslint-enable @typescript-eslint/no-var-requires */
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/scripts/junit2html.js]: Do we need the no-var-requires rule?

File: packages/patternfly-4/react-integration/scripts/server.js [hunk 10]
```diff
@@ -1,10 +1,13 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
 const cypress = require('cypress');
 const LocalWebServer = require('local-web-server');
+/* eslint-enable @typescript-eslint/no-var-requires */
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/scripts/server.js]: Do we need the no-var-requires rule?

File: packages/patternfly-4/react-integration/webpack.config.js [hunk 11]
```diff
@@ -1,3 +1,4 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/webpack.config.js]: Do we need the no-var-requires rule?

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/BackgroundImageDemo/BackgroundImageDemo.tsx (hunk 2 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/BackgroundImageDemo/BackgroundImageDemo.tsx]: Yep we're removing this now - thanks for calling this out!

File: packages/patternfly-4/react-integration/scripts/htmlReport.js [hunk 12]
```diff
@@ -1,12 +1,13 @@
-var xmlDocument = require('xmldoc');
+import { XmlDocument } from 'xmldoc';
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/scripts/htmlReport.js]: Do we need require here?

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/AppToolbar/AppToolbar.tsx [hunk 13]
```diff
@@ -10,7 +10,7 @@ import {
 } from '@patternfly/react-core';
 import CogIcon from '@patternfly/react-icons/dist/js/icons/cog-icon';
 
-const isDropdownOpen: boolean = false;
+const isDropdownOpen = false;
```
[the remark this answers is not in the document]
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/AppToolbar/AppToolbar.tsx]: This was due to [no-inferrable-types](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-inferrable-types.md), but I've disabled this rule and reverted the change as I think being more explicit is helpful in our case.

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/AboutModal/AboutModalDemo.tsx [hunk 14]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { AboutModal, Button, TextContent, TextList, TextListItem } from '@patternfly/react-core';
-import brandImg from './brandImg.svg';
+const brandImg = './brandImg.svg';
```
[the remark this answers is not in the document]
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/AboutModal/AboutModalDemo.tsx]: There was an error thrown with the import, `Cannot find module './brandImg.svg'.` . Looking at AboutModal where it's used, the `brandImgSrc` prop expects a string that's the URL of where to find the resource whereas we were passing in the actual SVG file.

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ApplicationLauncherDemo/ApplicationLauncherDemo.tsx [hunk 15]
```diff
@@ -18,7 +18,7 @@ export class ApplicationLauncherDemo extends React.Component<null, ApplicationLa
         isOpen
       });
     };
-    this.onSelect = event => {
+    this.onSelect = () => {
```
[the remark this answers is not in the document]
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ApplicationLauncherDemo/ApplicationLauncherDemo.tsx]: Thanks for the feedback - reverted

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/LoginPageDemo/LoginPageDemo.tsx (hunk 7 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/LoginPageDemo/LoginPageDemo.tsx]: Disabled rule and removed comment

File: packages/patternfly-4/react-integration/scripts/junit2html.js (hunk 9 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/scripts/junit2html.js]: Disabled rule and removed comment

File: packages/patternfly-4/react-integration/webpack.config.js (hunk 11 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/webpack.config.js]: Disabled rule and removed comment

File: packages/patternfly-4/react-integration/scripts/server.js (hunk 10 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/scripts/server.js]: Disabled rule and removed comment

File: packages/patternfly-4/react-integration/cypress/plugins/index.js [hunk 16]
```diff
@@ -10,6 +10,7 @@
 
 // This function is called when a project is opened or re-opened (e.g. due to
 // the project's config changing)
+// eslint-disable-next-line @typescript-eslint/no-var-requires
 const webpack = require('@cypress/webpack-preprocessor');
```
[the remark this answers is not in the document]
Comment (evwilkin) [reply packages/patternfly-4/react-integration/cypress/plugins/index.js]: Disabled rule and removed comment

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/AvatarDemo/AvatarDemo.tsx [hunk 17]
```diff
@@ -1,6 +1,6 @@
 import { Avatar, AvatarProps } from '@patternfly/react-core';
 import React, { Component } from 'react';
-import avatarImg from '../../../assets/images/imgAvatar.svg';
+const avatarImg = '../../../assets/images/imgAvatar.svg';
```
[the remark this answers is not in the document]
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/AvatarDemo/AvatarDemo.tsx]: Error was being thrown that the path could not be found.  Looking at `Avatar` component, the `src` prop where this is being used expects a URL string rather than an SVG file.

File: packages/patternfly-4/react-integration/scripts/htmlReport.js (hunk 12 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/scripts/htmlReport.js]: Updated to require XmlDocument

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ApplicationLauncherDemo/ApplicationLauncherFavoritesDemo.tsx (hunk 1 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ApplicationLauncherDemo/ApplicationLauncherFavoritesDemo.tsx]: Added this back in

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/OverflowMenuDemo/OverflowMenuDemo.tsx (hunk 8 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/OverflowMenuDemo/OverflowMenuDemo.tsx]: Added argument back in

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/BrandDemo/BrandDemo.tsx (hunk 3 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/BrandDemo/BrandDemo.tsx]: This was throwing an error that the path could not be resolved, and the `Brand` component `src` prop expects a URL string

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/DropdownDemo/DropdownDemo.tsx (hunk 5 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/DropdownDemo/DropdownDemo.tsx]: Added argument back in

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ContextSelectorDemo/ContextSelectorDemo.tsx (hunk 4 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ContextSelectorDemo/ContextSelectorDemo.tsx]: Added argument back in

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/FormSelectDemo/FormSelectDemo.tsx (hunk 6 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/FormSelectDemo/FormSelectDemo.tsx]: Added argument back in

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ApplicationLauncherDemo/ApplicationLauncherDemo.tsx [hunk 18]
```diff
@@ -18,7 +18,8 @@ export class ApplicationLauncherDemo extends React.Component<null, ApplicationLa
         isOpen
       });
     };
-    this.onSelect = event => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    this.onSelect = _event => {
```
Comment (ddonahue007) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ApplicationLauncherDemo/ApplicationLauncherDemo.tsx]: Curious here, if the '_event' var is not being used is there a reason we would not just use  the '() =>' notation here instead of applying the ignore to the linting rule? I could be missing why we are doing this so its more a question.  Happens more than once in this file. I can update each one I find once we determine if we want to leave it as is or make a change.

Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ApplicationLauncherDemo/ApplicationLauncherDemo.tsx]: Thanks for the call-out - these were updated per [Dan's comments](https://github.com/patternfly/patternfly-react/pull/3718#discussion_r378345721) above

Comment (dlabrecq) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ApplicationLauncherDemo/ApplicationLauncherDemo.tsx]: The underscore just eliminates the linter warning. In the case of a demo, it's desirable to show the expected variables.

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/TableDemo/TableActionsDemo.tsx [hunk 19]
```diff
@@ -72,16 +72,16 @@ export class TableActionsDemo extends React.Component<TableProps, TableState> {
...
+          // eslint-disable-next-line no-console
           console.log(`clicked on Some action, on row ${rowId} of type ${rowData.type}`)
       },
       {
         title: 'Another action',
-        // tslint:disable-next-line:no-shadowed-variable
-        onClick: (event, rowId, rowData, extra) =>
-          // tslint:disable-next-line:no-console
+        // eslint-disable-next-line no-shadow
+        onClick: (event, rowId, rowData) =>
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/TableDemo/TableActionsDemo.tsx]: Should we restore the `extra` prop in the functions above? Perhaps just rename it as _extra to avoid the warning?

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/OverflowMenuDemo/OverflowMenuDemo.tsx [hunk 20]
```diff
@@ -87,7 +87,7 @@ export class OverflowMenuDemo extends React.Component {
     });
   };
 
-  onAdditionalOptionsSelect = event => {
+  onAdditionalOptionsSelect = () => {
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/OverflowMenuDemo/OverflowMenuDemo.tsx]: Should we restore the `event` props in the functions above? Perhaps just rename it as _event to avoid the warning?

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/DropdownDemo/DropdownDemo.tsx [hunk 21]
```diff
@@ -62,13 +62,14 @@ export class DropdownDemo extends React.Component<{}, DropdownState> {
         isActionOpen
       });
     };
-    this.onActionSelect = event => {
+    this.onActionSelect = () => {
```
Comment (dlabrecq) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/DropdownDemo/DropdownDemo.tsx]: Should we restore the `event` props in the functions above? Perhaps just rename it as `_event` to avoid the warning?

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/TableDemo/TableActionsDemo.tsx (hunk 19 above)
Comment (evwilkin) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/TableDemo/TableActionsDemo.tsx]: Added

**Comment by evwilkin** [comment]
Added function arguments back in throughout demos where they were removed due to not being used

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/DropdownDemo/DropdownDemo.tsx [hunk 22]
```diff
@@ -46,7 +46,8 @@ export class DropdownDemo extends React.Component<{}, DropdownState> {
         isOpen
       });
     };
-    this.onSelect = event => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    this.onSelect = _event => {
```
Comment (jschuler) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/DropdownDemo/DropdownDemo.tsx]: or change to no params `() => {}`

Comment (dlabrecq) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/DropdownDemo/DropdownDemo.tsx]: Suggest we keep the params to show what is expected here, considering this is a demo
